### PR TITLE
Improvements for HwMon support

### PR DESCRIPTION
**Pull request by bjackman**
- Clean up some naming and use of the devlib API 
- Simplify configuring the channels (remove conf['conf'])
- Update devlib for bugfix used by subsequent commit
- Improve default channel_map so you don't need to specify it.

### CODE REVIEW

File: libs/utils/energy.py [hunk 1]
```diff
@@ -130,11 +130,31 @@ def __init__(self, target, conf=None, res_dir=None):
         self._log.info('Scanning for HWMON channels, may take some time...')
         self._hwmon = devlib.HwmonInstrument(self._target)
 
-        # record the HWMon channels
-        self._channels = conf.get('channel_map', {
-            'LITTLE': self._target.little_core.upper(),
-            'big': self._target.big_core.upper()
-        })
+        # Decide which channels we'll collect data from.
+        # If the caler provided a channel_map, require that all the named
```
Comment (jlelli) [inline libs/utils/energy.py]: s/caler/caller/ :-)

File: libs/utils/energy.py [hunk 2]
```diff
@@ -130,11 +130,31 @@ def __init__(self, target, conf=None, res_dir=None):
...
+        # channels exist.
+        # Otherwise, try using the big.LITTLE core names as channel names.
+        # If they dont' match, just collect all available channels.
```
Comment (jlelli) [inline libs/utils/energy.py]: s/dont'/don't/

**Comment by bjackman** [comment]
Rebased & fixed typos, thanks :)

**Comment by jlelli** [comment]
Looks good to me.
Tested on Juno.

**Comment by bjackman** [comment]
Thanks a lot for the test.

**Pull request merged**